File: implementations/hannoi_java_rec/Hannoi_rec.java

public class Hannoi_rec {

     static void move(int n, int from, int to, int via) {
        if (n >= 1)  
        {  
          move(n - 1, from, via, to);
          System.out.println("Move disk "+n +" from pole " + from + " to pole " + to);
          move(n - 1, via, to, from);
        }
     }

    public static void main(String[] args) {
      int disk =  (args.length >= 1) ? Integer.parseInt(args[0]) : 10 ; 
      move(disk, 1,2,3);
    }
 }

File: implementations/hannoi_python_rec/hannoi_rec.py
import sys 

def hanoi(ndisks, frome, to,via):
    if ndisks:
        hanoi(ndisks-1, frome,via,to)
        #print("Move disk %d from pole %d to pole %d" % (ndisks,frome,to))
        hanoi(ndisks-1,via,to,frome)
 

def main():
   if len(sys.argv) > 1:
      for arg in sys.argv[1:]:
         n = eval(arg)
         hanoi(n, 1, 2, 3)
   else:
         hanoi(10,1,2,3)

main()